src/analysis/processing/qgsalgorithmrasterize.cpp

```diff
@@ -150,6 +151,72 @@ QVariantMap QgsRasterizeAlgorithm::processAlgorithm( const QVariantMap &paramete
   int height { yTileCount * tileSize };
   int nBands { transparent ? 4 : 3 };
 
+  int64_t totalTiles = 0;
+  for ( auto &layer : std::as_const( mMapLayers ) )
+  {
+    if ( QgsRasterLayer *rasterLayer = qobject_cast<QgsRasterLayer *>( ( layer.get() ) ) )
+    {
+      if ( const QgsProviderMetadata *metadata = rasterLayer->providerMetadata() )
+      {
+        QVariantMap details = metadata->decodeUri( rasterLayer->source() );
+        if ( details.contains( QStringLiteral( "url" ) ) && QUrl( details[QStringLiteral( "url" )].toString() ).host() == QStringLiteral( "tile.openstreetmap.org" ) )
```

Comment:
Seems like "a.tile.openstreetmap.org" / "b.tile.openstreetmap.org" also work for fetching tiles.

Lets go with:

```suggestion
        if ( details.contains( QStringLiteral( "url" ) ) && QUrl( details[QStringLiteral( "url" )].toString() ).host().endsWith( QStringLiteral( ".openstreetmap.org" ) ) )
```

And I'd suggest moving this logic to  `QgsRasterLayerUtils::isOpenStreetMapLayer(const QgsRasterLayer* layer ) ` so that it's easy to keep in sync across algorithms

Reply:
There are also hosts ending with `tile.osm.org` which should be considered -> https://www.openstreetmap.org/user/pnorman/diary/396125#hosts (just verified that they are still working)

Reply:
@nyalldawson , @pathmapper , useful points, thanks for raising this.